classes/Tools14.php

```diff
@@ -24,2244 +24,2241 @@
-         */
-        public static function getPath($id_category, $path = '', $linkOntheLastItem = false, $categoryType = 'products')
-        {
-            global $link, $cookie;
+    /**
+     * Encrypt password
+     *
+     * @param object $object Object to display
+     */
+    public static function getAdminToken($string)
+    {
+        return !empty($string) ? self::encrypt($string) : false;
+    }
+    public static function getAdminTokenLite($tab)
+    {
+        global $cookie;
+        return self::getAdminToken($tab.(int)Tab::getIdFromClassName($tab).(int)$cookie->id_employee);
+    }
 
-            if ($id_category == 1)
-                return '<span class="navigation_end">'.$path.'</span>';
+    /**
+     * Get the user's journey
+     *
+     * @param integer $id_category Category ID
+     * @param string $path Path end
+     * @param boolean $linkOntheLastItem Put or not a link on the current category
+     * @param string [optionnal] $categoryType defined what type of categories is used (products or cms)
+     */
+    public static function getPath($id_category, $path = '', $linkOntheLastItem = false, $categoryType = 'products')
+    {
+        global $link, $cookie;
 
-            $pipe = Configuration::get('PS_NAVIGATION_PIPE');
-            if (empty($pipe))
-                $pipe = '>';
+        if ($id_category == 1)
+            return '<span class="navigation_end">'.$path.'</span>';
 
-            $fullPath = '';
+        $pipe = Configuration::get('PS_NAVIGATION_PIPE');
+        if (empty($pipe))
+            $pipe = '>';
 
-            if ($categoryType === 'products')
-            {
-                $category = Db::getInstance()->getRow('
-			SELECT id_category, level_depth, nleft, nright
-			FROM '._DB_PREFIX_.'category
-			WHERE id_category = '.(int)$id_category);
+        $fullPath = '';
 
-                if (isset($category['id_category']))
-                {
-                    $categories = Db::getInstance()->ExecuteS('
-				SELECT c.id_category, cl.name, cl.link_rewrite
-				FROM '._DB_PREFIX_.'category c
-				LEFT JOIN '._DB_PREFIX_.'category_lang cl ON (cl.id_category = c.id_category)
-				WHERE c.nleft <= '.(int)$category['nleft'].' AND c.nright >= '.(int)$category['nright'].' AND cl.id_lang = '.(int)($cookie->id_lang).' AND c.id_category != 1
-				ORDER BY c.level_depth ASC
-				LIMIT '.(int)$category['level_depth']);
-
-                    $n = 1;
-                    $nCategories = (int)sizeof($categories);
-                    foreach ($categories AS $category)
-                    {
-                        $fullPath .=
-                            (($n < $nCategories OR $linkOntheLastItem) ? '<a href="'.self::safeOutput($link->getCategoryLink((int)$category['id_category'], $category['link_rewrite'])).'" title="'.htmlentities($category['name'], ENT_NOQUOTES, 'UTF-8').'">' : '').
-                            htmlentities($category['
```

Comment:
This function looks very complicated. Can it be a template that we copy & paste with parametres, such as twig or smarty file?